lib/electric_slide/call_queue.rb

```diff
@@ -62,7 +62,7 @@ def available_agent_summary
     # @return {Agent}
     def checkout_agent
       agent = @strategy.checkout_agent
-      agent.presence = :busy
+      agent.presence = :busy if agent
```

Comment:
I made more changes than just that line, there is also a change to the connection method to requeue the "queued_call" if the agent if nil. 

Have a look at this line: https://github.com/adhearsion/electric_slide/blob/develop/lib/electric_slide/agent_strategy/longest_idle.rb#L26

Nothing is protecting against shifting an empty free_agents queue. If that queue is empty then we will return nil on checkout_agent. We should probably add nil checks on the return agent method too if there aren't already.